app/models/user.rb

```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+  has_many :comments, through: :posts
+
+  validates :name, presence: true
```

Comment:
The user model must have a method that returns the 3 most recent posts as part of the project requirement. Kindly implement that method. 👍